src/DataProtection/EntityFrameworkCore/src/EntityFrameworkCoreXmlRepository.cs

```diff
@@ -33,6 +33,10 @@ public EntityFrameworkCoreXmlRepository(IServiceProvider services, ILoggerFactor
 
         _logger = loggerFactory.CreateLogger<EntityFrameworkCoreXmlRepository<TContext>>();
         _services = services ?? throw new ArgumentNullException(nameof(services));
+
+        // Access DataProtectionKey properties dynamically to trigger DynamicallyAccessedMembers on the type
+        // and preserve the unused DataProtectionKey.Id property.
+        _ = typeof(DataProtectionKey).GetProperties();
```

Comment:
Why like this?

Reply:
This will work, but you can probably put a DynamicDependencyAttribute on the constructor instead to make it more explicit.

But this of course doesn't address the trimming warning generated at the reflection callsite (wherever this thing is reflection-consumed).

Reply:
> Why like this? Instead of DAM.

This + DAM is required. See linked comment in the issue description.

Reply:
> This will work, but you can probably put a DynamicDependencyAttribute on the constructor instead to make it more explicit.

Which constructor do you mean?

> But this of course doesn't address the trimming warning generated at the reflection callsite (wherever this thing is reflection-consumed).

Unfortunately EntityFrameworkCore isn't annotated for trimming yet. Once it is, this workaround should be able to be removed. See comment at https://github.com/dotnet/aspnetcore/issues/43187#issuecomment-1211385450

Reply:
> Which constructor do you mean?

The constructor that this comment is attached to. Putting `[DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(DataProtectionKey))]` on it will instruct trimming to consider all public properties on `DataProtectionKey` to be reflection-accessed when this constructor is kept.

Reply:
```
[DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(DataProtectionKey))]
public EntityFrameworkCoreXmlRepository(IServiceProvider services, ILoggerFactory loggerFactory) { .. }
```

Reply:
Got it. It's been a minute since I've thought deeply about trimming. DynamicDependency is a better solution here.

Thanks.